src/battle_script_commands.c

```diff
@@ -14758,6 +14758,8 @@ static void Cmd_handleballthrow(void)
 
     if (gBattleTypeFlags & BATTLE_TYPE_TRAINER)
     {
+        if (B_TRY_CATCH_TRAINER_BALL >= GEN_4)
+            AddBagItem(gLastUsedItem, 1);
```

Comment:
Why is it added back instead of preventing the removal?

Reply:
It felt like the easiest place to insert it tbh, but I can also try to go a step further into the code to prevent it from being removed in the first place if you prefer

Reply:
Yeah, I think it would be nicer to prevent the removal.

Reply:
Done!

Reply:
Ngl, I'm now questioning myself if it's really nicer that way. Maybe a 3rd person could chime in or you make the final decision.